cli/tests/integration/lsp_tests.rs

```diff
@@ -3480,6 +3480,8 @@ fn lsp_diagnostics_refresh_dependents() {
     .write_request::<_, _, Value>("shutdown", json!(null))
     .unwrap();
   client.write_notification("exit", json!(null)).unwrap();
+  // wait for the published diagnostics of both requests
+  std::thread::sleep(std::time::Duration::from_millis(250));
```

Comment:
>> Flaky test most likely caused by this PR (extra diagnostic when none was expected):
>
> Yeah, what is happening is sometimes now the diagnostics are coming out of "order", because now the lint category is decoupled from the typescript ones, the "last 3" aren't always correct...
>
> Given the change, I would say it would safe to only look at the "last" diagnostics and ensure they are empty. (i == queue_len - 1).

@kitsonk this PR actually doesn't decouple the diagnostics. It just gets the lint and module diagnostics publishing right away.

I looked at the test code a bit more and I think this sleep should stop the test from being flaky. I'm not proud of it though 🙈. Let me know if you have another suggestion.